### DISCUSSION TITLE
"error initialising SDL: No available video device" on Debian 9

### DISCUSSION THREAD

**Discussion by cg00001** [redeclipse/base#1157] [Help and Support]
Can't run RE on my debian laptop with amd drivers I get "error initialising SDL: No available video device".

Anyone?

**Comment by proneon267** [comment]
See this: https://askubuntu.com/questions/443053/could-not-initialize-sdl-no-available-video-device-dont-move although it's for ubuntu 12.04 but it might still work. And https://wiki.libsdl.org/FAQLinux#I_get_the_error:_.22no_video_devices_available.22

**Comment by cg00001** [comment]
Besides building SDL, I have tried all other solutions mentioned above including reinstalling (via sudo apt install --reinstall) all libsdl1.2 and libsdl2 libs with no success. Maybe I should build SDL and try again.

Maybe it's the radeon drivers, I think I have played RE several months back with the fglrx drivers, but I'm not sure -too many games.

**Comment by no-lex** [comment]
What does  `lspci -k | grep --EA2 "VGA"` return? It should tell you which kernel driver you're on. Also, is RE the only 3d game that doesn't work (or does stuff like glxgears work and return a reasonable framerate)?

**Comment by cg00001** [comment]
I found out it might be some library (ies) that depend on the newest MESA that Debian stable doesn't support yet (I use backport repo and for example to get the libSDL2-dev library it needs that new MESA I can't install without breaking stable -installing from sid).
Most 3d games including the glxgears app run fine, here's some more info:

`k@kozidaware:~$ lspci -k | grep "VGA"
00:01.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Kabini [Radeon HD 8210]
k@kozidaware:~$ lspci | grep VGA
00:01.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Kabini [Radeon HD 8210]
k@kozidaware:~$ lsmod | grep "kms\|drm"
drm_kms_helper        196608  2 amdgpu,radeon
drm                   475136  8 gpu_sched,drm_kms_helper,amdgpu,radeon,ttm
k@kozidaware:~$ find /dev -group video
/dev/media0
/dev/video1
/dev/video0
/dev/fb0
/dev/dri/card0
/dev/dri/renderD128
k@kozidaware:~$ cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-4.18.0-0.bpo.1-amd64 root=UUID=8a914cf3-ceb4-4516-8b12-3a8e26416737 ro quiet acpi_osi=Linux acpi_backlight=video
k@kozidaware:~$ find /etc/modprobe.d/
/etc/modprobe.d/
/etc/modprobe.d/oss-compat.conf
/etc/modprobe.d/dkms.conf
/etc/modprobe.d/mdadm.conf
/etc/modprobe.d/amd64-microcode-blacklist.conf
/etc/modprobe.d/intel-microcode-blacklist.conf
/etc/modprobe.d/radeon-kms.conf
k@kozidaware:~$ cat /etc/modprobe.d/*kms*
# modprobe information used for DKMS modules
#
# This is a stub file, should be edited when needed,
# used by default by DKMS.
options radeon modeset=1

k@kozidaware:~$ ls /etc/X11/xorg.conf
ls: cannot access '/etc/X11/xorg.conf': No such file or directory
k@kozidaware:~$ glxinfo | grep -i "vendor\|rendering"
direct rendering: Yes
server glx vendor string: SGI
client glx vendor string: Mesa Project and SGI
    Vendor: X.Org (0x1002)
OpenGL vendor string: X.Org
k@kozidaware:~$ grep LoadModule /var/log/Xorg.0.log
[    66.658] (II) LoadModule: "glx"
[    67.416] (II) LoadModule: "ati"
[    67.539] (II) LoadModule: "radeon"
[    67.590] (II) LoadModule: "modesetting"
[    67.757] (II) LoadModule: "fbdev"
[    67.828] (II) LoadModule: "vesa"
[    68.072] (II) LoadModule: "fbdevhw"
[    68.133] (II) LoadModule: "fb"
[    68.349] (II) LoadModule: "dri2"
[    68.349] (II) LoadModule: "glamoregl"
[    74.797] (II) LoadModule: "ramdac"
[    78.516] (II) LoadModule: "libinput"
[    79.194] (II) LoadModule: "synaptics"
`

**Comment by cg00001** [comment]
I managed to install all .dev and mesa files needed with aptitude, purged red eclipse and reinstalled. Same error message. It must be the radeon drivers -fglrx worked fine.

Let me know if there is anything I can do to get the game working, thanks.
